ovirt/resource_ovirt_affinity_group.go

```diff
@@ -0,0 +1,425 @@
+				MustBuild())
+		attributeUpdate = true
+	}
+
+	vmRuleBuilder := ovirtsdk4.NewAffinityRuleBuilder()
+	vmRuleUpdate := false
+	if d.HasChange("vm_positive") {
+		vmRuleBuilder.Positive(d.Get("vm_positive").(bool))
+		vmRuleUpdate = true
+	}
+	if d.HasChange("vm_enforcing") {
+		vmRuleBuilder.Enforcing(d.Get("vm_enforcing").(bool))
+		vmRuleUpdate = true
+	}
+	if d.HasChange("vm_list") {
+		vmRuleBuilder.Enabled(len(d.Get("vm_list").([]interface{})) > 0)
+		vmRuleUpdate = true
+	}
+	if vmRuleUpdate {
+		group.VmsRule(vmRuleBuilder.MustBuild())
+		attributeUpdate = true
+	}
+
+	hostRuleBuilder := ovirtsdk4.NewAffinityRuleBuilder()
+	hostRuleUpdate := false
+	if d.HasChange("host_positive") {
+		hostRuleBuilder.Positive(d.Get("host_positive").(bool))
+		hostRuleUpdate = true
+	}
+	if d.HasChange("host_enforcing") {
+		hostRuleBuilder.Enforcing(d.Get("host_enforcing").(bool))
+		hostRuleUpdate = true
+	}
+	if d.HasChange("host_list") {
+		hostRuleBuilder.Enabled(len(d.Get("host_list").([]interface{})) > 0)
+		hostRuleUpdate = true
+	}
+	if hostRuleUpdate {
+		group.HostsRule(hostRuleBuilder.MustBuild())
+		attributeUpdate = true
+	}
+
+	if attributeUpdate {
+		log.Printf("[DEBUG] Updating %#v", group.MustBuild())
+		_, err := conn.SystemService().
+			ClustersService().
+			ClusterService(d.Get("cluster_id").(string)).
+			AffinityGroupsService().
+			GroupService(d.Id()).
+			Update().
+			Group(group.MustBuild()).
+			Send()
+		if err != nil {
+			log.Printf("[DEBUG] Error updating affinity group (%s): %s", d.Id(), err)
+			return err
+		}
+	}
+
+	if d.HasChange("vm_list") {
+		vmsService := conn.SystemService().
+			ClustersService().
+			ClusterService(d.Get("cluster_id").(string)).
+			AffinityGroupsService().
+			GroupService(d.Id()).
+			VmsService()
+
+		if err := updateVmList(vmsService, d.Get("vm_list").([]interface{})); err != nil {
+			return err
+		}
+	}
+
+	if d.HasChange("host_list") {
+		hostsService := conn.SystemService().
+			ClustersService().
+			ClusterService(d.Get("cluster_id").(string)).
+			AffinityGroupsService().
+			GroupService(d.Id()).
+			HostsService()
+
+		if err := updateHostList(hostsService, d.Get("host_list").([]interface{})); err != nil {
+			return err
+		}
+	}
+
+	return resourceOvirtAffinityGroupRead(d, meta)
+}
+
+func resourceOvirtAffinityGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*ovirtsdk4.Connection)
+	_, err := conn.SystemService().
+		ClustersService().
+		ClusterService(d.Get("cluster_id").(string)).
+		AffinityGroupsService().
+		GroupService(d.Id()).
+		Remove().
+		Send()
+	if err != nil {
+		if _, ok := err.(*ovirtsdk4.NotFoundError); ok {
+			return nil
+		}
+		return err
+	}
+	return nil
+}
+
+func stringInSlice(a string, list []interface{}) bool {
+	for _, b := range list {
+		if b.(string) == a {
+			return true
+		}
+	}
+	return false
+}
+
+func updateVmList(vmsService *ovirtsdk4.AffinityGroupVmsService, vmList []interface{}) error {
```

Comment:
this should accept affinityGroupsGroupService

Reply:
I cannot find that object in go-ovirt?